coderd/httpmw/recover.go

```diff
@@ -0,0 +1,40 @@
+package httpmw
+
+import (
+	"context"
+	"net/http"
+	"runtime/debug"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/coderd/httpapi"
+)
+
+func Recover(log slog.Logger) func(h http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			defer func() {
+				err := recover()
```

Comment:
Hmm the stack trace should get us to where we want to get but i'll print the contents of `r` as well.

Reply:
You're probably right, I had a slightly different thought but after thinking about it again, it'd be rare enough that it's not worth considering.

My thought: I was thinking more along the lines that, since we're using `xerrors` (which attaches stack traces), it would be possible for the error (if passed to `panic`) to contain a stack trace from other goroutines. It could provide extra context regarding what went wrong.

But we'd have to explicitly `panic(err)` for that to be of any help, which we should avoid anyway.